game/resources.cpp

```diff
@@ -506,28 +506,47 @@ std::unique_ptr<ProtoMesh> Resources::implLoadMeshMain(std::string name) {
       mdhName = name;
     FileExt::assignExt(mdhName,"MDH");
 
-    const auto* entry = Resources::vdfsIndex().find(mdhName);
-    if(entry==nullptr)
-      throw std::runtime_error("failed to open resource: " + mdhName);
+    try {
+      const auto* entry = Resources::vdfsIndex().find(mdhName);
+      if(entry==nullptr)
+        throw std::runtime_error("failed to open resource: " + mdhName);
 
-    zenkit::ModelHierarchy mdh;
-    auto reader = entry->open_read();
-    mdh.load(reader.get());
+      zenkit::ModelHierarchy mdh;
+      auto reader = entry->open_read();
+      mdh.load(reader.get());
 
-    std::unique_ptr<Skeleton> sk{new Skeleton(mdh,anim,name)};
-    std::unique_ptr<ProtoMesh> t;
+      std::unique_ptr<Skeleton> sk{new Skeleton(mdh,anim,name)};
+      std::unique_ptr<ProtoMesh> t;
 
-    if(const auto* entry = Resources::vdfsIndex().find(mesh)) {
-      auto reader = entry->open_read();
+      if(const auto* entry = Resources::vdfsIndex().find(mesh)) {
+        auto reader = entry->open_read();
 
-      zenkit::ModelMesh mdm {};
-      mdm.load(reader.get());
-      t.reset(new ProtoMesh(mdm,std::move(sk),name));
+        zenkit::ModelMesh mdm {};
+        mdm.load(reader.get());
+        t.reset(new ProtoMesh(mdm,std::move(sk),name));
+        }
+      else
+        t.reset(new ProtoMesh(mdh,std::move(sk),name));
+
+      return t;
       }
-    else
-      t.reset(new ProtoMesh(mdh,std::move(sk),name));
+      catch(...) {
```

Comment:
In opengothic I'm using exceptions only for exceptional cases (aka errors). Based on your PR description, absence of `mdh` is not abnormal case, but just a 'special' one.

It would be better to rewrite `try-catch` part as:
```c+++
if(const auto* entry = Resources::vdfsIndex().find(mdhName)) {
  // MDH path
  } else {
  // MDL fallback
  }
```

Reply:
I've adapted the code accordingly now to not needlessly throw exceptions around.